src/components/implementation/simple_mc_udp_server/simple_mc_udp_server_no_lwip/main.c

```diff
@@ -4,18 +4,79 @@
 #include <mc.h>
 #include <simple_udp_stack.h>
 #include <netshmem.h>
+#include <sched.h>
 
 static int fd;
+static volatile thdid_t init_thd = 0;
 
 void
 cos_init(void)
 {
 	shm_bm_objid_t objid;
 	struct netshmem_pkt_buf *obj;
-
+	init_thd = cos_thdid();
 	/* create current component's shmem */
 	netshmem_create();
+}
+void
+cos_parallel_init(coreid_t cid, int init_core, int ncores)
+{
+	if (cid == 0) {
+		return;
+	}
+
+	if (cos_compid() == 6 && cid != 1) {
+		return;
+	}
+	if (cos_compid() == 7 && cid != 2) {
+		return;
+	}
+	if (cos_compid() == 8 && cid != 3) {
+		return;
+	}
+	if (cos_compid() == 9 && cid != 4) {
+		return;
+	}
+	if (cos_compid() == 10 && cid != 5) {
+		return;
+	}
+	if (cos_compid() == 11 && cid != 6) {
+		return;
+	}
+	if (cos_compid() == 12 && cid != 7) {
+		return;
+	}
+	if (cos_compid() == 13 && cid != 8) {
+		return;
+	}
+	if (cos_compid() == 14 && cid != 9) {
+		return;
+	}
+	if (cos_compid() == 15 && cid != 10) {
+		return;
+	}
+	if (cos_compid() == 16 && cid != 11) {
+		return;
+	}
+	if (cos_compid() == 17 && cid != 12) {
+		return;
+	}
+	if (cos_compid() == 18 && cid != 13) {
+		return;
+	}
+	if (cos_compid() == 19 && cid != 14) {
+		return;
+	}
+	if (cos_compid() == 20 && cid != 15) {
+		return;
+	}
+	if (cos_compid() == 21 && cid != 16) {
+		return;
+	}
 
+	if (init_thd != cos_thdid()) {
+		netshemem_move(init_thd, cos_thdid());
```

Comment:
First I noticed this name. `netshemem` looks like "net she memory", which isn't very informative. If it is supposed to be for shared memory, then the `e` is throwing off the name.